### DISCUSSION TITLE
Using Floating-UI with Suggestions

### DISCUSSION THREAD

**Discussion by tom-coupe-sb** [ueberdosis/tiptap#4608] [General Questions] [answered]
Hey peeps, 

I'm defining a custom mention plugin, and I would prefer to use Floating UI with React rather than tippy for the suggestion popover. However, as far as I can tell, you need to be able to pass a `ref` into your reference React component for it to work as a positioning anchor. `SuggestionProps` supplies `clientRect` which can be used for positioning in tippy, as well as `decorationNode`, whose purpose is unclear; neither of these work with the standard Floating UI React syntax. Any workarounds people are aware of?

**Comment by kurisLei** [comment]
```lang=html
<n-popover trigger="hover">
  <template #trigger>
    <n-button>Click</n-button>
  </template>
  <span>some thing</span>
</n-popover>
```

I also encountered the same problem. In the Naive UI I use, the Popover component requires a Trigger node, but decorationNode does not seem to be used as a mounting node, so I can only manually locate it through the clientRect method.

**Comment by nperez0111** [comment, marked as answer]
In Tiptap v3 we will be moving to Floating UI: https://github.com/ueberdosis/tiptap/pull/5398
